ratpack-core/src/main/java/ratpack/http/client/internal/HttpChannelKey.java

```diff
@@ -17,25 +17,33 @@
 package ratpack.http.client.internal;
 
 import io.netty.channel.EventLoop;
+import ratpack.api.Nullable;
 import ratpack.exec.ExecController;
 import ratpack.exec.Execution;
 
 import java.net.URI;
 import java.time.Duration;
+import java.util.Objects;
 
 final class HttpChannelKey {
 
   final boolean ssl;
   final int port;
   final String host;
 
+  final ProxyInternal proxy;
```

Comment:
I added ProxyInternal to the HttpChannelKey because Proxy settings influence how the client connects to the destination. This was partly to make them accessible to the ChannelPoolHandler.channelCreated where the proxy Netty handler is created.

I tried moving the handler creation to RequestActionSupport, but by the time RequestActionSupport gets the channel, it's too late to add the proxy handler(s).

A positive consequence of this is that proxy connections, including their tunnels, are pooled. Thus, when using proxies, the client does not always have to establish a new tunnel to the destination.

The proxy settings are only added to the HttpChannelKey if there are proxy settings. Thus, is no additional memory overhead in the common case where proxies are not used.